crates/router/src/core/payments/helpers.rs

```diff
@@ -1648,7 +1648,7 @@ pub fn check_force_psync_precondition(
             | storage_enums::AttemptStatus::Authorized
             | storage_enums::AttemptStatus::Started
             | storage_enums::AttemptStatus::Failure
-    ) && connector_transaction_id.is_some()
+    ) || connector_transaction_id.is_some()
```

Comment:
This would lead to unnecessary connector calls, in case of terminal statuses too

Reply:
This is not the location that we need to change the condition. This needs to be left unchanged

The or logic needs to be present here:
`crates/router/src/core/payments/operations/payment_status.rs:340:33`